autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterCommonService.java

```diff
@@ -178,12 +178,14 @@ public AuthorizationResourceType getSupportedAuthorizationResourceType() {
     @Override
     @Retryable(value = NotFoundException.class, maxAttempts = 3, backoff = @Backoff(delay = 5000))
     public String getResourceCrnByResourceName(String clusterName) {
+        // All controller methods annotated with @CheckPermissionByResourceName will forcefully sync a cluster to periscope for a successful authz check
         return getClusterByCrnOrName(NameOrCrn.ofName(clusterName)).getStackCrn();
     }
 
     @Override
     public Optional<String> getEnvironmentCrnByResourceCrn(String resourceCrn) {
-        return Optional.ofNullable(getClusterByCrnOrName(NameOrCrn.ofCrn(resourceCrn)).getEnvironmentCrn());
+        Optional<Cluster> clusterByCrn = clusterService.findOneByStackCrnAndTenant(resourceCrn, restRequestThreadLocalService.getCloudbreakTenant());
+        return Optional.of(clusterByCrn.map(Cluster::getEnvironmentCrn).orElse(resourceCrn));
```

Comment:
@horadla23 requesting your input on this change. If I understand correctly, this method is only called for API endpoints annotated with `@CheckPermissionByResourceCrn`. If a periscope cluster has not been persisted into the DB yet, would this logic cause any issues in API authorization? 
Context: Due to simultaneous invocations of APIs with `@CheckPermissionByResourceCrn`, duplicate entity records are being populated into the DB

Reply:
this is called in case of `@CheckPermissionByEnvironmentCrn` or if there is hierarchical authz (which is true in case of datahub resource), where we are not just checking stack crn but corresponding env crn too if needed.
If the cluster is not persisted, then authz probably will fail since the framework won't find the corresponding cluster.
If that is the case, we need to find another way to get corresponding cluster.

Reply:
`.orElse(resourceCrn)` this is something which won't work, if we give resourceCrn back instead of environmentCrn, it will never work, authz will fail in case of every environmentCrn check

Reply:
you can simply return with this: `return clusterByCrn.map(Cluster::getEnvironmentCrn);`
if there is no environmentCrn, no need to do the extra authz check for environment for no reason.

Reply:
Thanks, updated this